core/src/main/java/tech/tablesaw/io/csv/CsvReaderAdapter.java

```diff
@@ -0,0 +1,32 @@
+package tech.tablesaw.io.csv;
+
+import com.univocity.parsers.common.IterableResult;
+import com.univocity.parsers.common.ParsingContext;
+import com.univocity.parsers.csv.CsvFormat;
+import com.univocity.parsers.csv.CsvParser;
+import com.univocity.parsers.csv.CsvParserSettings;
+
+import tech.tablesaw.io.UnicodeBOMInputStream;
+
+public interface CsvReaderAdapter {
+
+    String[] readNext();
+
+    void skipLines(int linesToSkip);
+
+    static CsvReaderAdapter createWithOptions(UnicodeBOMInputStream ubis, char separator) {
+        
+        CsvParserSettings parserSettings = new CsvParserSettings();
+        parserSettings.setLineSeparatorDetectionEnabled(true); // TODO discuss
```

Comment:
What is the question to discuss?  
I'm not sure this something we need? I don't recall seeing many/any file loads fail because of a non-standard line-end character.